### DISCUSSION TITLE
Yarn Issue on Install

### DISCUSSION THREAD

**Discussion by camthoms** [mastodon/mastodon#20689] [Q&A] [answered]
When I'm trying to run:

"yarn install --pure-lockfile"

I'm getting this error:

Unknown Syntax Error: Unsupported option name ("--pure-lockfile")

**Comment by ineffyble** [comment]
Your globally installed version of Yarn is Yarn Modern (Yarn 2/3), whereas this project is still using Yarn Classic.

You can try `yarn set version classic`, downgrading your globally installed version to Yarn 1, or if you just do a `yarn install` without `--pure-lockfile` you will probably be okay.

**Reply by camthoms** [reply]
Setting to classic still doesn't let me use --pure-lockfile. Runing yarn install I get this error:

![image](https://user-images.githubusercontent.com/118034249/202458161-b155c899-7bed-458b-9b3d-4da981598f66.png)


Which leads to this error when I compile the assets:
![image](https://user-images.githubusercontent.com/118034249/202459308-e7af3c95-7efb-4329-9a13-f9ab491e59c6.png)

**Reply by ineffyble** [reply]
Those logs suggest setting to classic didn't work, that's yarn 3 format logs 🤔 

Are you able to try downgrading yarn globally to latest yarn 1.x?

**Reply by camthoms** [reply]
> Those logs suggest setting to classic didn't work, that's yarn 3 format logs 🤔
> 
> Are you able to try downgrading yarn globally to latest yarn 1.x?

Yeah I set it there in root, but I guess because the /home/mastodon/.yarnrc.yml file was pointing to the 3.2.4 folder it wanted to use that. I deleted the folder like a fool and now I can't use yarn commands because the path doesn't exist

**Comment by camthoms** [comment, marked as answer]
For those that might find this down the line: I ended up re-doing everything on Ubuntu 20.04 instead of 22.04 like the documentation recommends, who knew???
